namesys/publisher.go

```diff
@@ -110,6 +124,30 @@ var IpnsRecordValidator = &record.ValidChecker{
 	Sign: true,
 }
 
+func IpnsSelectorFunc(k key.Key, vals [][]byte) (int, error) {
+	var recs []*pb.IpnsEntry
+	for _, v := range vals {
+		e := new(pb.IpnsEntry)
+		err := proto.Unmarshal(v, e)
+		if err != nil {
+			// Probably dont need to fully error out, other records might be valid
+			return 0, err
+		}
+	}
+
+	var best_seq uint64
+	var best_i int
+
+	for i, r := range recs {
+		if r.GetSequence() > best_seq {
+			best_seq = r.GetSequence()
+			best_i = i
+		}
+	}
+
+	return best_i, nil
```

Comment:
This function should just sort them and return them sorted. see the records spec for why.